File: be_camphub/src/main/java/org/camphub/be_camphub/dto/response/notification/NotificationResponse.java
package org.camphub.be_camphub.dto.response.notification;

import java.util.UUID;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class NotificationResponse {
    UUID id;
    UUID receiverId;
    UUID senderId;
    String type;
    String title;
    String content;
    String referenceType;
    UUID referenceId;
    boolean isRead;
    String createdAt;
}

File: fe_camphub/src/libs/core/dto/response/notification.response.ts
import { NotificationType, ReferenceType } from "@/libs/core/constants";

export interface NotificationResponse {
    id: string;
    receiverId: string | null;
    senderId: string;
    type: NotificationType;
    title: string;
    content: string;
    referenceType: ReferenceType;
    referenceId: string;
    isRead: boolean;
    isBroadcast: boolean;
    createdAt: string;
}